docs/adr/0003-use-of-devcontainer-directory.md

```diff
@@ -0,0 +1,55 @@
+
+The `.devcontainer` directory is one of the [standard locations](https://code.visualstudio.com/docs/devcontainers/create-dev-container#_create-a-devcontainerjson-file) for the `devcontainer.json` file, which is the primary configuration file for dev containers. The other location option is put this json file in the root of the project directory.
+
+Elements of configuration of the dev container can exist in both the research template docker image, and others in the `.devcontainer` directory (e.g. `devcontainer.json` and supporting scripts/files). Updates to `.devcontainer` in the research template will require a mechanism to push these updates to research repos derived from this template. Updates to the research template docker image will require building and publishing of this image to a container registry. We have implemented a [CI action](https://github.com/opensafely-core/research-template-docker/actions/workflows/main.yml) to build and publish the research template docker image on merge to the `main` branch and scheduled weekly.
+
+One element of the dev container configuration which cannot easily be set in the docker image build is that of setting working directories for R/RStudio to the [workspace folder](https://containers.dev/implementors/json_reference/#variables-in-devcontainerjson) which is (by default) not known until container creation time.
+
+
+New repositories created from the research template will contain the latest `.devcontainer` directory as of their creation. There is a need to deploy the `.devcontainer` directory to existing research repos, both for initial addition of dev containers/codespaces functionality to the repo, and for future configuration updates. Options for deployment of the template `.devcontainer` directory to existing research repos include:
+* manual deployment by researchers
+* manual deployment by Team REX
+* automated deployment
+
+Options for the degree of ownership of this directory include: 
+* Team REX for both the template and in deployed versions in researcher repos
+* Team REX for the template and research repo owners for deployed versions in research repos
+
+We wish to minimise the effort of Team REX in supporting codespaces, and so wish to minimise divergence from a standard configuration. However, we acknowledge there may be circumstances in which researchers have a legitimate need for custom configuration. Any deployment of updates to the `.devcontainer` directory to a research repo should ideally be respectul of this.
+
+
+## Decision
+
+We will keep any dev container configuration in research-template repo within the `.devcontainer` directory where possible. Putting it in this subdirectory helps separate this configuration from researcher code.
```

Comment:
I think this should come second. We've agreed to keep any _remaining_ dev container configuration in `.devcontainer`.